elide-core/src/main/java/com/yahoo/elide/core/ResourceLineage.java

```diff
@@ -66,6 +66,18 @@ public ResourceLineage(ResourceLineage sharedLineage, PersistentResource next, S
         return list == null ? Collections.emptyList() : list;
     }
 
+    /**
+     *  Returns the immediate parent resource if one exists.
+     * @return the parent or null if there is no parent.
+     */
+    public PersistentResource getParent() {
+        if (resourcePath.size() == 0) {
```

Comment:
Empty check is often faster than counting the size of a list
```suggestion
        if (resourcePath.isEmpty()) {
```